packages/addon/public/api/TBProMenu/implementation.js

```diff
@@ -0,0 +1,526 @@
+            }
+            if (attributes) {
+              Object.keys(attributes).forEach((key) =>
+                element.setAttribute(key, attributes[key])
+              );
+            }
+
+            if (id) {
+              element.id = id;
+            }
+
+            if (i18nId) {
+              element.textContent = this._localize(i18nId);
+            } else if (text) {
+              element.textContent = text;
+            }
+
+            return element;
+          },
+
+          /**
+           * Generate the app menu header button for TBPro. This will generate
+           * the button with no text.
+           *
+           * @returns {HTMLElement}
+           */
+          _generateHeaderButton() {
+            const toolbarButton = this._createElement({
+              type: 'toolbarbutton',
+              classes: [
+                'subviewbutton',
+                'subviewbutton-iconic',
+                'tbpro-header-button',
+              ],
+              attributes: { closemenu: 'none' },
+              xul: true,
+            });
+
+            const divider = this._createElement({
+              type: 'div',
+              classes: ['tbpro-divider'],
+            });
+
+            const text = this._createElement({
+              type: 'span',
+              classes: ['tbpro-menu-item-text'],
+            });
+            const wrapper = this._createElement({
+              type: 'span',
+              classes: ['tbpro-header-content'],
+            });
+            const boldText = this._createElement({
+              type: 'span',
+              classes: ['tbpro-menu-item-bold-text'],
+            });
+
+            wrapper.appendChild(text);
+            wrapper.appendChild(boldText);
+
+            toolbarButton.appendChild(wrapper);
+            toolbarButton.appendChild(divider);
+
+            return toolbarButton;
+          },
+
+          /**
+           * Get the translated string if available of the i18n message specified.
+           * @param {string} string - The message id to get translation for.
+           * @returns {string} The translated message.
+           */
+          _localize(string) {
+            return context.extension.localeData.localizeMessage(string);
+          },
+
+          // An event to inform the WebExtension the menu entry has been clicked.
+          onCommand: new ExtensionCommon.EventManager({
+            context,
+            name: 'TBProMenu.onCommand',
+            register(fire) {
+              function callback(event, window, action) {
+                // Let the event return the windowId of the window where the menu
+                // item was clicked along with the action name for context.
+                return fire.async(
+                  context.extension.windowManager.getWrapper(window).id,
+                  action
+                );
+              }
+
+              emitter.on('menu-item-clicked', callback);
```

Comment:
We don't need to use our own emitter here, we can use the one from the extension. This makes sure we're using the right instance of the emitter across API calls.